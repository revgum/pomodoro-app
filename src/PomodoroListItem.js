// @flow
import * as React from "react";
import "./PomodoroListItem.css";
import * as Utils from "./utils";
import type { PauseState } from "./types";
import PauseList from "./PauseList";

// material-ui
import { withStyles } from "material-ui/styles";
import Card, { CardContent, CardHeader } from "material-ui/Card";
import Delete from "@material-ui/icons/Delete";
import IconButton from "material-ui/IconButton";
import red from "material-ui/colors/red";

type Props = {
  classes: any,
  deletePomodoro: Function,
  id: number,
  isPaused: boolean,
  pauses: Array<PauseState>,
  pausedElapsed: number,
  secondsRemaining: number,
  visible: boolean
};

const styles = theme => ({
  card: {
    marginBottom: theme.spacing.unit,
    padding: 0
  },
  cardContent: {
    padding: 12
  },
  deleteAction: {
    color: red[500]
  },
  pos: {
    marginBottom: 12
  }
});

class PomodoroListItem extends React.Component<Props> {
  render() {
    return (
      <li className={this.classes()}>
        <Card className={this.props.classes.card}>
          <CardHeader
            className={this.props.classes.cardHeader}
            action={
              <IconButton>
                <Delete
                  className={this.props.classes.deleteAction}
                  onClick={this.deletePomodoro}
                />
              </IconButton>
            }
            title={this.headerTitle()}
          />
          <CardContent className={this.props.classes.cardContent}>
            <PauseList
              isPaused={true}
              pauses={this.props.pauses}
              pausedElapsed={this.props.pausedElapsed}
            />
          </CardContent>
        </Card>
      </li>
    );
  }

  headerTitle = () => {
    if (this.props.secondsRemaining <= 0) return "Complete";
    return `${Utils.formattedTime(this.props.secondsRemaining)} remaining`;
  };

  classes = () => {
    return this.props.visible ? "" : "hidden";
  };

  deletePomodoro = () => {
    this.props.deletePomodoro(this.props);
  };
}

export default withStyles(styles)(PomodoroListItem);
