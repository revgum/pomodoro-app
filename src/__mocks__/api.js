import { isNullOrUndefined } from "util";

const mockInitialState = {
  test: "test"
};

export async function initialState(error) {
  if (!error) return mockInitialState;
  throw new Error("mocked error");
}

export async function save(timer_state, error) {
  if (!error) return timer_state;
  throw new Error("mocked save error");
}

export async function fetchAll(error) {
  if (!error) return [];
  throw new Error("mocked fetchAll error");
}

export async function destroy(timer_state, error) {
  if (!error) return timer_state;
  throw new Error("mocked destroy error");
}
