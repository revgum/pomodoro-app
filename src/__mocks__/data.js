export const pomodoros = [
  {
    id: 1,
    visible: true,
    classes: {},
    isPaused: false,
    isStarted: true,
    seconds: 0,
    minutes: 25,
    pauses: [],
    pausedElapsed: 0,
    secondsElapsed: 0,
    secondsRemaining: 25 * 60,
    timer: null
  }
];

export const lots_of_pomodoros = [
  pomodoros[0],
  Object.assign({}, pomodoros[0], { id: 2 }),
  Object.assign({}, pomodoros[0], { id: 3 }),
  Object.assign({}, pomodoros[0], { id: 4 })
];
