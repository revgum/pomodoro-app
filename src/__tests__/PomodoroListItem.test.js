import React from "react";
import PomodoroListItem from "../PomodoroListItem";
import { createShallow, createRender } from "material-ui/test-utils";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import * as mockData from "../__mocks__/data";
import shadows from "material-ui/styles/shadows";
import Card, { CardContent, CardHeader } from "material-ui/Card";

describe("<PomodoroListItem />", () => {
  let shallow;
  let wrapper;
  let instance;
  beforeAll(() => {
    shallow = createShallow();
    wrapper = shallow(<PomodoroListItem deletePomodoro={jest.fn()} />).dive();
    instance = wrapper.instance();
  });

  it("passes a smoke test", () => {
    const wrapper = shallow(<PomodoroListItem />);
  });

  it("captures a snapshot", () => {
    const tree = renderer.create(<PomodoroListItem />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("deletes this pomodoro", () => {
    let mounted = mount(<PomodoroListItem deletePomodoro={jest.fn()} />);
    mounted.find("button").simulate("click");
    //TODO: test that the button was clicked and deletePomodoro was called
  });
  it("displays a complete pomodoro", () => {
    let mounted = mount(
      <PomodoroListItem deletePomodoro={jest.fn()} secondsRemaining={0} />
    );
    let header = mounted.find(CardHeader);
    expect(header.find({ title: "Complete" })).toExist();
  });
  it("displays a pomodoro with time remaining", () => {
    let mounted = mount(
      <PomodoroListItem deletePomodoro={jest.fn()} secondsRemaining={1} />
    );
    let header = mounted.find(CardHeader);
    expect(header.find({ title: "00:01 remaining" })).toExist();
  });
});
