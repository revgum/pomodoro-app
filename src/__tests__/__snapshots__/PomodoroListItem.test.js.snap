// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PomodoroListItem /> captures a snapshot 1`] = `
<li
  className="hidden"
>
  <div
    className="MuiPaper-root-6 MuiPaper-elevation2-10 MuiPaper-rounded-7 MuiCard-root-5 PomodoroListItem-card-1"
  >
    <div
      className="MuiCardHeader-root-33"
    >
      <div
        className="MuiCardHeader-content-36"
      >
        <span
          className="MuiTypography-root-39 MuiTypography-headline-44 MuiCardHeader-title-37"
        >
          NaN:NaN remaining
        </span>
      </div>
      <div
        className="MuiCardHeader-action-35"
      >
        <button
          className="MuiButtonBase-root-69 MuiIconButton-root-63"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-68"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-72 PomodoroListItem-deleteAction-3"
              color={undefined}
              focusable="false"
              onClick={[Function]}
              viewBox="0 0 24 24"
            >
              <g>
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                />
              </g>
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-78"
          />
        </button>
      </div>
    </div>
    <div
      className="MuiCardContent-root-85 PomodoroListItem-cardContent-2"
    />
  </div>
</li>
`;
